sympy/sets/sets.py

```diff
@@ -1968,6 +1968,14 @@ def __lt__(self, other):
             raise TypeError("Invalid comparison of set with %s" % func_name(other))
         return self.is_proper_subset(other)
 
+    def __eq__(self, other):
```

Comment:
I just checked on live.sympy.org and master branch, `FiniteSet(Rational(3, 10), Rational(1, 5)) == FiniteSet(0.3, 0.2)`  does not work on master either.
And I think, it is because:

In [`Basic.__eq__`, it checks for `_hashable_content()`](https://github.com/sympy/sympy/blob/master/sympy/core/basic.py#L328) of both the operands.

And [`_hashable_content()` returns a `frozenset`](https://github.com/sympy/sympy/blob/master/sympy/sets/sets.py#L1941)

So basically, we are equating `frozenset([Rational(3, 10), Rational(1, 5)]) == frozenset([0.2, 0.3])` which is false